### ISSUE TITLE
Pokemon Omega Ruby & Alpha Sapphire Issues

### ISSUE THREAD

**Issue by ybha4** [citra-emu/citra#2041]
Hey guys, now that CRO has been merged, Pokemon is now officially supported by Citra (_hears sighs of developers_).

Seriously though, there are a couple of issues with Pokemon:

1) Frequently in the game, there are a lot of black lines that appear and possibly, some depth issues?

2) The game crashes frequently. Is this still missing areas within CRO implementation? Or is this a separate issue within Citra? Also, the crashes don't appear to happen at consistent points in the game. ie sometimes it will crash in the beginning. Once replayed, it will continue passed that point and might crash at another random point in the game. The crashing appears to be completely random.

To replicate, open Pokemon OR or AS and play the first 5 minutes of the game. You will see what I'm talking about.

Specs: 
Windows 7
i7 (3.4ghz)
8gb RAM
Nvidia

Note: This is different to #2026 and #2028 as those crashes occur at specific points in the game. Mine are random it seems.
![pokemon issue 1](https://cloud.githubusercontent.com/assets/21698970/18191265/718f9702-710c-11e6-8dd3-b4916934cd4c.png)

Enjoy :)

**Comment by yuriks** [comment]
The issue with the wrong outlines seems to be caused by improper management of the stencil buffer in the texture cache: it'll sometimes not be cleared even though the depth buffer.

That issue and possibly other crashes might be caused by #2053.

**Comment by wwylele** [comment]
I did a test about crash/freeze, and get an interesting result.

I tested Pokemon OR (JAP version, language Japanese), with a save from others (who complained to me that the save was "cursed" and freeze frequently). I am on the JIT branch, though, but I believe it doesn't matter. 

The test is: start the game, walk into grass, start a battle and finish it without any crash/freeze. This test was performed several times.

To find the cause of the freeze, I change the code for a half among all samples. The change is: stub [the result code of NWM_UDS::Initialize](https://github.com/citra-emu/citra/blob/master/src/core/hle/service/nwm_uds.cpp#L100) with 0xC9411002 instead of SUCCESS. (the error code is get from hw, which I believe to be "network disabled")

And the result of all samples:


|Code changed?|Total samples|Failed|Pass|
|---|---|---|---|
|No|10|9|1|
|Yes|10|0|10|

:smile: I guess we get a direction to work on.

**Comment by ybha4** [comment]
On an interesting note, ever since the JIT was merged, the game seems to crash more often than before. As for whether this is simply a matter of chance, I am unsure (but it seems to crash twice as frequently now).

**Comment by hcorion** [comment]
@ybha4 I'm pretty sure that is entirely coincidental. I seem to be able to make it farther (from the very beginning) with the JIT than before.

**Comment by Subv** [comment]
Please test again with #2218

**Comment by Dragios** [comment]
Black lines are gone but the thick lines behind the starter Pokémon still remain though.

**Comment by freiro** [comment]
@Dragios could you update the first post and provide a new screenshoot please?

**Comment by Dragios** [comment]
This is what happen when scaled resolution is turned off initially
![Thick line present](https://cloud.githubusercontent.com/assets/18496868/20667589/01151faa-b5a5-11e6-907c-f8de9b765079.png)

This is the result of turning on *scaled resolution* and subsequent toggle on/off *scaled resolution*
![Thick line absent](https://cloud.githubusercontent.com/assets/18496868/20667846/6f89d844-b5a6-11e6-8d54-6e9137e07f1e.png)

Ignore the red circle on the screenshot as I thought is a bug because of partial missing edge marking. After checking with the reference 3DS it also suffers from the same issue or is intended.

This log appear when reaching this scene
```
HW.GPU <Critical> video_core\renderer_opengl\gl_shader_gen.cpp:AppendSource:58: Unhandled texture type 1
HW.GPU <Critical> video_core\renderer_opengl\gl_shader_gen.cpp:AppendSource:58: Unhandled texture type 1
HW.GPU <Critical> video_core\renderer_opengl\gl_shader_gen.cpp:AppendSource:58: Unhandled texture type 1
HW.GPU <Critical> video_core\renderer_opengl\gl_shader_gen.cpp:AppendSource:58: Unhandled texture type 1
HW.GPU <Critical> video_core\renderer_opengl\gl_shader_gen.cpp:AppendSource:58: Unhandled texture type 1
HW.GPU <Critical> video_core\renderer_opengl\gl_shader_gen.cpp:AppendSource:58: Unhandled texture type 1
```

**Comment by JayFoxRox** [comment]
Could we please split this issue into 2 or more parts?
(Possibly new issues altogether and close this one)

We have multiple different disucssions here..
- At least 2 independent crashes (one addressed by wwylele in #2126 another one unsolved).
- At least 3 different visual issues (lines when there shouldn't be any [fixed in #2218 ], missing lines [false positive], bars in background [might be related to texture type 1])

Each issue title should describe the wrong behaviour you see (which is not the case here). And information such as logs might hint at bugs but they are not necessarily responsible (if they turn out to be the culprit we'll label the issue for you).
Meta-issues (which group issues per game) like this one should avoided. They make it impossible to label issues and it makes it harder to find out what's being discussed.

**Comment by ybha4** [comment]
On a side note, once #2260 is merged, all of the issues addressed above (except the thick bars when choosing the starter pokemon) are fixed. The game no longer crashes randomly at all and is very much playable (though I have only tested about 5 - 10 mins of the game).
